### PR TITLE
Implement account registration in the wizard

### PR DESCRIPTION
**Pull request by juliusknorr**
Based on #9 

2017-07-11-221227_750x507_scrot

2017-07-11-221235_750x507_scrot

ToDo:
- [ ] Implement pending account type for registrations
- [ ] Frequently check for registration status
- [ ] Setup HttpCredentials with app password after registration is complete

### CODE REVIEW

File: src/gui/ocsregistrationjob.h [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    void verify(const QString &username, const QString &displayname, const QString &email);
+    void createRegistration(const QString &username, const QString &displayname, const QString &email, const QString &password);
+    void status(const QString &clientSecret);
+
+signals:
+    /**
+     * Result of the OCS request
+     * The value parameter is only set if this was a put request.
+     * e.g. if we set the password to 'foo' the QVariant will hold a QString with 'foo'.
+     * This is needed so we can update the share objects properly
```
Comment (rullzer) [inline src/gui/ocsregistrationjob.h:32]: Copy paste :wink:

File: src/gui/wizard/owncloudproviderlistpage.h [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    Q_OBJECT
+public:
+    OwncloudProviderListPage(QWidget *parent);
+    ~OwncloudProviderListPage();
+    int nextId() const;
+    bool isComplete() const;
+    void initializePage();
+public slots:
+    void toggleFreePlans(bool state);
+    void setCountry(QString current);
```
Comment (rullzer) [inline src/gui/wizard/owncloudproviderlistpage.h:27]: use http://doc.qt.io/qt-5/qlocale.html#Country-enum ?

**Review by rullzer — CHANGES_REQUESTED** [review]
I'll have a more detailed look soon

**Comment by juliusknorr** [comment]
Closing in favour of https://github.com/nextcloud/gsoc_client/pull/17

**Pull request closed without merging**